stacks-core/src/utils.rs

```diff
@@ -111,6 +112,40 @@ impl From<StacksAddress> for PrincipalData {
 	}
 }
 
+impl TryFrom<String> for PrincipalData {
+	type Error = StacksError;
+
+	fn try_from(value: String) -> Result<Self, Self::Error> {
+		let parts: Vec<&str> = value.split('.').collect();
+
+		match parts.len() {
+			1 => {
+				let address = StacksAddress::try_from(parts[0])?;
+				Ok(Self::Standard(StandardPrincipalData::from(address)))
+			}
+			2 => {
+				let address = StacksAddress::try_from(parts[0])?;
+				let contract_name = parts
+					.get(1)
```

Comment:
Let's stick to either index accessing or `.get()` but not both. If index is out of bounds it'll just crash but `.get()` allows us to return a nicer error message to the caller.

Reply:
how can this be out of bounds if we check the length?

Reply:
I meant when you access with an index that can crash, get will return an option.